msal/src/Microsoft.Identity.Client/Features/ConfidentialClient/IConfidentialClientApplicationWithCertificate.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Component to be used for confidential client applications like Web Apps/API.
+    /// This component supports Subject Name + Issuer Authentication
+    /// </summary>
+    public interface IConfidentialClientApplicationWithCertificate
+    {
+        /// <summary>
+        /// Acquires token from the service for the confidential client using the client credentials flow. See https://aka.ms/msal-net-client-credentials
+        /// This method enables application developers to achieve easy certificates roll-over
+        /// in Azure AD: this method will send the public certificate to Azure AD
+        /// along with the token request, so that Azure AD can use it to validate the subject name based on a trusted issuer policy.
+        /// This saves the application admin from the need to explicitly manage the certificate rollover
+        /// (either via portal or powershell/CLI operation)
+        /// </summary>
+        /// <param name="scopes">Array of scopes requested for resource</param>
+        /// <returns>Authentication result containing application token for the requested scopes</returns>
+        Task<AuthenticationResult> AcquireTokenForClientWithCertificateAsync(IEnumerable<string> scopes);
+
+        /// <summary>
+        /// Acquires token from the service for the confidential client using the client credentials flow. See https://aka.ms/msal-net-client-credentials
+        /// This method attempts to look up valid access token in the cache unless<paramref name="forceRefresh"/> is true
+        /// This method enables application developers to achieve easy certificates roll-over
+        /// in Azure AD: this method will send the public certificate to Azure AD
+        /// along with the token request, so that Azure AD can use it to validate the subject name based on a trusted issuer policy.
+        /// This saves the application admin from the need to explicitly manage the certificate rollover
+        /// (either via portal or powershell/CLI operation)
+        /// </summary>
+        /// <param name="scopes">Array of scopes requested for resource</param>
+        /// <param name="forceRefresh">If TRUE, API will ignore the access token in the cache and attempt to acquire new access token using client credentials</param>
+        /// <returns>Authentication result containing application token for the requested scopes</returns>
+        Task<AuthenticationResult> AcquireTokenForClientWithCertificateAsync(IEnumerable<string> scopes, bool forceRefresh);
+
+        /// <summary>
+        /// Acquires token using On-Behalf-Of flow.
```

Comment:
>Acquires token using On-Behalf-Of flow. [](start = 12, length = 39)

you can add the aka.ms link here as well as below

Reply:
#Resolved

Reply:
link? same below btw.

---
In reply to: [208423187](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1124#discussion_r208423187) [](ancestors = 208423187)